torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2364,6 +2363,39 @@ std::tuple<Tensor, Tensor> atan2_backward(const Tensor& grad, const Tensor& self
             output_mask[0] ? grad * other * recip : Tensor(),
             output_mask[1] ? grad * -self * recip : Tensor() };
 }
+std::tuple<Tensor, Tensor, Tensor> attn_backward(
+    const Tensor& grad_o,
+    const Tensor& grad_a,
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v,
+    const Tensor& a) {
+  auto grad_q = Tensor();
+  auto grad_k = Tensor();
+  auto grad_v = Tensor();
+
+  if (!grad_o.defined() && !grad_a.defined()) {
+    return std::tuple<Tensor, Tensor, Tensor>{grad_q, grad_k, grad_v};
+  }
+  auto tanh_x_2 = (1 - a.square());
+  if (!grad_o.defined()) {
+    grad_q = (grad_a.mul(tanh_x_2)).matmul(k);
+    grad_k = (grad_a.mul(tanh_x_2)).transpose(0, 1).matmul(q);
+    grad_v = at::zeros_like(v);
```

Comment:
You can also just return a `Tensor()` here, which constructs an undefined tensor